docs/pages/indexing.md

```diff
@@ -1,18 +1,325 @@
+the member functions of the af::array class. These functions allow you to
+reference one or a range of elements from the original array.
+
+Here we will demonstrate some of the ways you can use indexing in ArrayFire and
+discuss ways to minimize the memory and performance impact of these operations.
+
+Lets start by creating a new 4x4 matrix of floating point numbers:
+
+\snippet test/index.cpp index_tutorial_1
+
+The ArrayFire library is column-major so the resulting A array will
+look like this:
+
+\f[
+\begin{bmatrix}
+    0 & 4 & 8 & 12 \\
+    1 & 5 & 9 & 13 \\
+    2 & 6 & 10 & 14 \\
+    3 & 7 & 11 & 15
+\end{bmatrix}
+\f]
+
+This is a two dimensional array so we can access the first element of this
+matrix by passing `0,0` into the parenthesis operator of the af::array.
+
+\snippet test/index.cpp index_tutorial_first_element
+
+\f[ A(2, 3) = [ 14 ] \f]
+
+We can also access the array using linear indexing by passing in one value. Here
+we are accessing the fifth element of the array.
+
+\snippet test/index.cpp index_tutorial_fifth_element
+
+\f[ A(5) = [ 5 ] \f]
+
+\note Normally you want to avoid accessing individual elements of the array like this
+for performance reasons.
+
+Negative indexing values will index from the end of the array. For example, the
+value negative one and negative two(-2) will return the last and second to last
+element of the array. ArrayFire provides the end alias for this which also
+allows you to index the last element of the array.
+
+\snippet test/index.cpp index_tutorial_negative_indexing
+
+## Indexing slices and subarrays
+
+You can access regions of the array via the af::seq and af::span objects. The
+span objects allows you to select the entire set of elements across a particular
+dimension/axis of an array. For example, we can select the third column of the
+array by passing span as the first argument and 2 as the second argument to the
+parenthesis operator.
+
+\snippet test/index.cpp index_tutorial_third_column
+
+\f[
+A(span, 2) =
+\begin{bmatrix}
+    8 \\
+    9 \\
+    10 \\
+    11
+\end{bmatrix}
+\f]
+
+You can read that as saying that you want all values across the first dimension,
+but only from index 2 of the second dimension.
+
+You can access the second rows by passing (1, span) to the array
+
+\snippet test/index.cpp index_tutorial_second_row
+
+\f[ A(1, span) = [ 1, 5, 9, 13 ] \f]
+
+You can use the af::seq (short for sequence) object defines a range of values
+when indexing. For example if you wanted to get the first two columns, you would
+access the array by passing span for the first argument and seq(2) as the second
+argument.
+
+\snippet test/index.cpp index_tutorial_first_two_columns
+
+\f[
+A(span, seq(2)) =
+\begin{bmatrix}
+     0 & 4 \\
+     1 & 5 \\
+     2 & 6 \\
+     3 & 7
+\end{bmatrix}
+\f]
+
+The af::seq object can also be used to define ranges that are not continuous. There are
+three constructors for af::seq.
```

Comment:
Second sentence stops abruptly. I think it should end like the following
```
There are three constructors for af::seq that can help create non-continuous ranges(af::seq objects)`

Reply:
Updated that section to:

```
There are three constructors for af::seq. 

* af::seq(N): Defines a range between 0 and N-1
* af::seq(begin, end) Defines a range between begin and end inclusive
* af::seq(begin, end, step) defines a range between begin and end striding by step values

The last constructor that can help create non-continuous ranges. For example,
you can select the second and forth(last) rows by passing (seq(1, end, 2), span)
to the indexing operator.
```

Reply:
`The af::seq object can also be used to define ranges that are not continuous.` Is this sentence gone now ?

Reply:
Yeah, I moved that message to the paragraph after the list.